src/libstd/sys/windows/os.rs

```diff
@@ -100,16 +104,41 @@ impl Iterator for Env {
                     Some(p) => p,
                     None => continue,
                 };
-                return Some((
-                    OsStringExt::from_wide(&s[..pos]),
-                    OsStringExt::from_wide(&s[pos+1..]),
-                ))
+                return Some((p, pos, len as usize))
```

Comment:
Could this return `s`? That seems like it may be tad bit safer than returning raw pointers